### DISCUSSION TITLE
IE_GPU tests fail.

### DISCUSSION THREAD

**Discussion by urbalke** [openvinotoolkit/openvino#20903] [Q&A]
Hello,

I'm poking around and I got to stage where Ive wanted to check all of the tests on my local machine. Unfortunately IE_GPU.* tests fail. Im currently on Ubuntu machine with nvidia graphics. How should I configure my environment?

**Comment by Ganesh-403** [comment]
@urbalke One thing to clarify first: are you using an **Intel GPU** or only an **NVIDIA GPU**?

The `IE_GPU.*` tests exercise the **OpenVINO GPU plugin**, which is designed for **Intel GPUs** (integrated or discrete) using the Intel OpenCL/Level Zero runtime—not NVIDIA GPUs. :contentReference[oaicite:0]{index=0}

If your machine only has an NVIDIA GPU, I'd actually expect the GPU plugin tests to fail because the required Intel GPU device won't be available.

A couple of things to check:

1. What does:

   ```bash
   clinfo
   ```

   report?

2. Does OpenVINO detect any GPU device?

   ```cpp
   ov::Core core;
   for (const auto& d : core.get_available_devices())
       std::cout << d << std::endl;
   ```

   If no `GPU` device appears, the GPU plugin won't be able to run. OpenVINO only enumerates supported Intel GPU devices. :contentReference[oaicite:1]{index=1}

3. Are you intending to develop the GPU plugin itself, or are you simply trying to run the project's test suite? If it's the latter, you may need to disable or skip the `IE_GPU.*` tests on a machine without supported Intel GPU hardware.

If you do have an Intel GPU in addition to the NVIDIA card, could you share:

- `lspci | grep -Ei "VGA|3D"`
- `clinfo`
- the exact test failure output

That would help determine whether this is a driver/configuration issue or something else.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
